docs/formatter.qmd

```diff
@@ -6,4 +6,180 @@ editor:
+Having a formatter as companion significantly improves the process of writing code as you no longer have to think about style as much and just let it do that work for you.
+
+Second, it reduces friction when working in a team.
+By agreeing to use a formatter in a project, collaborators no longer have to discuss styling and layout issues.
+Code sent to you by a colleague will adhere to the standards that you're used to.
+Code review no longer has to be about style nitpicks and can focus on the substance of the changes instead.
+
+# How does Air decide how to format your code?
+
+Air tries to strike a balance between enforcing rigid rules and allowing authors some control over the layout.
+Our main source of styling rules is the [Tidyverse style guide](https://style.tidyverse.org).
+
+There is a trend among modern formatter of being *opionated*.
+Air certainly fits this trend and provides very few [configuration options](https://posit-dev.github.io/air/configuration.html), mostly the indent style (spaces versus tabs), the indent width, and the line width.
+However, Air also puts code authors in charge of certain aspects of the layout through the notion of **persisent line breaks**.
+
+In general, Air is in control of deciding where to put vertical space (line breaks) in your code.
+For instance if you write:
+
+``` r
+list(foo,
+bar)
+```
+
+Air will figure out that this expression fits on a single line without exceeding the line width.
+It will discard the line break and reformat to:
+
+``` r
+list(foo, bar)
+```
+
+However there are very specific places at which you can enforce a line break, i.e. make it persistent.
+
+-   Before the very first argument in a function call.
+    This:
+
+    ``` r
+    list(
+    foo, bar)
+    ```
+
+    gets formatted as:
+
+    ``` r
+    list(
+      foo,
+      bar
+    )
+    ```
+
+-   Before the very first right-hand side expression in a pipeline.
+    This:
+
+    ``` r
+    1 +
+    2 + 3
+    ```
+
+    gets formatted as:
+
+    ``` r
+    1 +
+      2 +
+      3
+    ```
+
+A persistent line break will never be removed by Air.
+But you can remove it manually.
+Taking the last example, if you join the first lines like this:
+
+``` r
+list(foo,
+  bar
+)
+
+1 + 2 +
+  3
+```
+
+Air will format as:
+
+``` r
+list(foo, bar)
+
+1 + 2 + 3
+```
+
+The goal of this feature is to strike a balance between being opionated and being flexible.
+
+# When does a formatter run?
+
+The most convenient way to run a formatter is when you save a file.
+This is not typically enabled by default so you will need to enable it manually (see the [Editors](editors.qmd) section).
+Once enabled, your code will be reformatted every time you save!
+This prevents your code from ever diverging from the standards set by Air.
+Another way of running the formatter in an IDE is when you paste code (this is enabled by the setting "Format on paste").
```

Comment:
I think I would leave this sentence out for now, since I feel like the main point of this section is:
- Learn how to format on save
- Learn that the CLI exists to format a whole project at once

Reply:
hmm... that makes sense for the website. For the disucssions page, it's probably still be useful in order to collect feedback on that feature.